onnx/external_data_helper.py

```diff
@@ -93,31 +97,39 @@ def set_external_data(tensor,  # type: TensorProto
             entry.value = str(v)
 
 
-def convert_model_to_external_data(model, all_tensors_to_one_file=True, location=None):
+def convert_model_to_external_data(model, all_tensors_to_one_file=True, location=None, size_threshold=1024):
     # type: (ModelProto, bool, Optional[Text]) -> None
     """
-    call to set all tensors as external data. save_model saves all the tensors data as external data after calling this function.
+    Call to set all tensors with raw data as external data. This call should preceed 'save_model'.
+    'save_model' saves all the tensors data as external data after calling this function.
     @params
     model: ModelProto to be converted.
     all_tensors_to_one_file: If true, save all tensors to one external file specified by location.
                              If false, save each tensor to a file named with the tensor name.
     location: specify the external file that all tensors to save to.
               If not specified, will use the model name.
+    size_threshold: Threshold for size of data. Only when tensor's data is >= the size_threshold
+    it will be converted to external data. To convert every tensor with raw data to external data set size_threshold=0.
     """
     if all_tensors_to_one_file:
         file_name = Text(uuid.uuid1())
         if location:
             file_name = location
         for tensor in _get_all_tensors(model):
-            set_external_data(tensor, file_name)
+            if tensor.HasField("raw_data") and sys.getsizeof(tensor.raw_data) >= size_threshold:
+                set_external_data(tensor, file_name)
```

Comment:
Shouldn't multiple tensors be stored at different offsets within the same file? I don't see any offsets here. How does this work? (Alternatively, if that feature is not being used, may be we can remove it for now until we can add a proper version.)

Reply:
This method simply marks the tensors to use external data. The offsets are properly added when the data is actually written to the file. See save_external_data line 175

Reply:
This is one reason to expose a single method that does both things. Otherwise, we have a fragile dependence between the two parts, with the model being in a weird non-standard representation in between the two steps ... if the user ends up doing other things in between the two steps, the potential for bugs increases.

Reply:
yes I agree... we should have a single method to do this... I will cover it as part of a new PR